### PR TITLE
Improve the documentation of Azure OpenAI

### LINKED ISSUE #15: Update the documentation to explain how Azure SDK for Java is used for OpenAI
Update the documentation to explain how Azure SDK for Java is used for OpenAI

> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [x] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

### PR DESCRIPTION
**Pull request by pmalarme**
## Purpose
* Improve the documentation of Azure OpenAI and the usage of the java client library

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Review the changes in root README.md

## What to Check
Verify that the following are valid
* OpenAI is better explained
* Link to OpenAI client for Java
* Link to prompt engineering documentation
* Explanation of where to find the different pieces in the code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,19 +77,38 @@ At the end of the deployment, you will see the URL of the front-end. Open the UR
...
 
 ![Architecture diagram](./assets/architecture-diagram.png)
 
 > This template provisions resources to an Azure subscription that you will select upon provisioning them. Please refer to the [Pricing calculator for Microsoft Azure](https://azure.microsoft.com/pricing/calculator/) and, if needed, update the included Azure resource definitions found in `infra/main.bicep` to suit your needs.
 
+## Azure OpenAI
+
+This sample application uses Azure OpenAI. It is not part of the automated deployment process. You will need to create an Azure OpenAI resource and configure the application to use it. Please follow the instructions in the [Azure OpenAI documentation](https://learn.microsoft.com/en-us/azure/ai-services/openai/overview#how-do-i-get-access-to-azure-openai) to get access to Azure OpenAI. This sample app was developed and tested using `gpt-4` model. You need the following information from the Azure OpenAI resource to configure the application:
```
Comment (aarthiem) [inline README.md]: Add a note for networking options or go with default ones?

File: README.md [hunk 2]
```diff
@@ -77,19 +77,38 @@ At the end of the deployment, you will see the URL of the front-end. Open the UR
...
+
+- `azureOpenAiApiKey` - Azure OpenAI API key
+- `azureOpenAiEndpoint` - Azure OpenAI endpoint
+- `azureOpenAiDeploymentId` - Azure OpenAI deployment ID of `gpt-4` model
```
Comment (aarthiem) [inline README.md:105]: add details for how to retrieve the open ai deployment id

**Review by aarthiem — COMMENTED** [review]
Thanks Pierre. have few comments, please review

File: README.md (hunk 2 above)
Comment (aarthiem) [reply README.md:105]: also, lets add links for responsible AI documentation

File: README.md (hunk 1 above)
Comment (pmalarme) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (pmalarme) [reply README.md:105]: Done both

**Review by aarthiem — APPROVED** [review]
Thanks Pierre!

**Pull request merged**